pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -573,6 +573,9 @@ private static void validateFieldConfigList(@Nullable List<FieldConfig> fieldCon
             Preconditions.checkArgument(!noDictionaryColumns.contains(columnName),
                 "FieldConfig encoding type is different from indexingConfig for column: " + columnName);
           }
+          Preconditions.checkArgument(fieldConfig.getNoDictionaryColumnCompressorCodec() == null,
+              "FieldConfig column compression codec is only supported for single value raw encoding type");
```

Comment:
(nit) Also add "Set compression codec to null for dictionary encoding type"

Reply:
done!